design/design.md

```diff
@@ -0,0 +1,112 @@
+Applications?
+mithril, peras, leios
+
+What cryptographic primitives will we implement?
+I propose that we implement the generic ALBA primitive to make it as applicable
+as possible.
+This means there is only a generic notion of "elements" and no notions of
+signatures, vrf, etc.
+This would be somewhat difficult to work with for all our three applications
+that really just need a weighted multisignature scheme.
+Hence, my second proposal is to use this generic ALBA implementation to implement
+a convenient multisignature scheme.
+It would also serve as a nice example of how the generic ALBA code can be used.
+
+Which ALBA algorithms to implement?
+For the centralized unweighted setting, there are four known algorithms to date:
+(basic) Telescope, Telescope with prehashing, Telescope with prehashing,
+repetitions and bounded DFS, and finally simple lottery.
+I propose that we only implement the third scheme -- Telescope with prehashing,
+repetitions and bounded DFS, and possibly the simple lottery.
+The basic Telescope is way slower and offers almost no advantage over the other
+schemes.
+Telescope with prehashing is only applicable when the number of honest prover's
+element n_p is large which is probably uncommon.
+Moreover, it offers no advantage over the third scheme except for a very tiny
+difference in the certificate size (integer v).
+As an optimization of the Telescope with prehashing, repetitions and bounded
+DFS, we could even eliminate this integer from the certificate if the
+number of allowed repetitions r = 1.
+Finally, the simple lottery, while having large proof sizes, offers very
+fast proving times which might be important for some applications.
+For the decentralized setting and weighted settings, we again have four
+possible algorithms:
+the three Telescope algorithms combined with the simple lottery, and the
+simple lottery itself.
+I propose we implement Telescope with prehashing, repetitions and bounded DFS,
```

Comment:
There seems to be some repetition here, or am I missing something?

Reply:
Lottery is repeated I think?

I think it would be simpler to keep lottery as decentralised hence we would have :
- Centralised (3 schemes): naive, prehashing, bounded
- Decentralised (4): only lottery, lottery + Centralised
Finally, we have "extensions": weighted and PRF.